build.gradle

```diff
@@ -1,90 +1,98 @@
-/*
- * This build file was generated by the Gradle 'init' task.
- *
- * This generated file contains a commented-out sample Java project to get you started.
- * For more details take a look at the Java Quickstart chapter in the Gradle
- * user guide available at https://docs.gradle.org/4.1/userguide/tutorial_java_projects.html
- */
-
-buildscript {
-    repositories {
-        mavenCentral()
-    }
-    dependencies {
-        classpath("org.springframework.boot:spring-boot-gradle-plugin:1.5.8.RELEASE")
-    }
-}
-
-plugins {
-    id 'org.springframework.boot' version '1.5.8.RELEASE'
-    id "com.gorylenko.gradle-git-properties" version "1.4.16"
-}
+import org.springframework.boot.gradle.plugin.SpringBootPlugin
 
-// Apply the java plugin to add support for Java
-apply plugin: 'java'
-version = "0.6.8"
+import java.time.Instant
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
 
-springBoot {
-    buildInfo()
+plugins {
+    id 'java'
+    id 'groovy'
+    id 'org.springframework.boot' version '2.1.8.RELEASE'
+    id 'io.spring.dependency-management' version '1.0.8.RELEASE'
+    id "com.gorylenko.gradle-git-properties" version "2.2.0"
+    id 'com.github.ben-manes.versions' version '0.25.0'
 }
 
+project.sourceCompatibility = 1.8
+project.targetCompatibility = 1.8
 
-sourceCompatibility = 1.8
+def today = Instant.now()
+group = 'uk.gov.justice.digital.delius'
 
-// Apply application plugin for easy application start
-apply plugin: 'application'
+def todayDate = LocalDate.now()
+def formatter = DateTimeFormatter.ofPattern('yyyy-MM-dd')
+version todayDate.format(formatter)
```

Comment:
Are we missing build number on here? If we have two builds on same day would they not have the same version label? Looks jar name in Circle only has date is that correct?

Reply:
Agreed, normally doesn't matter since we build docker containers instead.
Worth therefore grabbing the spring boot stuff and moving it up higher i.e. something like
```groovy
def todaysDate = LocalDate.now().format('yyyy-MM-dd')
version = System.getenv('CI') ? "${todaysDate}.${System.getenv('CIRCLE_BUILD_NUM')}" : todaysDate
```
and then reference `version` lower down instead